File: projeto/java/Proxy.java
import org.zeromq.ZMQ;

public class Proxy {
    public static void main(String[] args) {

        ZMQ.Context context = ZMQ.context(1);

        ZMQ.Socket xsub = context.socket(ZMQ.XSUB);
        xsub.bind("tcp://*:5557");

        ZMQ.Socket xpub = context.socket(ZMQ.XPUB);
        xpub.bind("tcp://*:5558");

        System.out.println("Proxy iniciado");

        ZMQ.proxy(xsub, xpub, null);
    }
}

File: projeto/python/Proxy.py
import zmq

def main():
    context = zmq.Context()

    xsub = context.socket(zmq.XSUB)
    xsub.bind("tcp://*:5557")

    xpub = context.socket(zmq.XPUB)
    xpub.bind("tcp://*:5558")

    print("Proxy iniciado")

    zmq.proxy(xsub, xpub, None)

if __name__ == "__main__":
    main()